_source/_posts/2022-07-27-python-okta.md

```diff
@@ -0,0 +1,191 @@
+
+Not every project needs all three of these things. You will generally need a token, you may not always need the other items. Sometimes you may require additional credentials when integrating with third party applications.
+
+## Configure the Okta CLI
+
+Let's get the Okta CLI installed. Okta has a great [resource page](https://cli.okta.com/) about the CLI and it contains installation instructions. We're going to do ours slightly differently from the instructions, so be sure to follow the steps here. 
+
+1. Install the CLI with the provided command:
+```bash
+brew install –cask oktadeveloper/tap/okta
+```
+
+2. Check and see if the CLI is accessible by trying out a command. If everything's working you'll see some information about how to use the CLI presented. 
+```bash
+okta –help
+```
+
+3. Next, we need to log in. We already have our Okta developer account, so in the Terminal, type:
+```bash
+okta login
+```
+
+4. Now we want to provide the credentials we are asked for. When asked for your organization, this is the URL that appears in your developer account in the upper right corner. Type out the entire URL, if the format is different the CLI will make you enter it again. You'll also need an API token. In your developer dashboard, from the choices on the left, click **Security** and choose **API**.
+
+5. On the API screen you can see your organization URL(s) and a tab called **Tokens**. Click the **Tokens** tab.
+
+6. On the Token page, you can manage and see all the tokens you've created. You can't see the token's value after creating it, so save the token information somewhere secure. Click **Create Token**.
+
+7. Follow the Create Token wizard to make your new token, and save the value to put in the CLI when prompted for credentials.
+
+After you've done this, you will have a developer account associated with the CLI. If you want to change it you can wipe the file and log in with a different Org URL and credentials. For now, let's leave it set up. 
+
+## Configure the project
+
+If you download the CLI, you can type `okta start flask` and follow the provided instructions to get the project set up. The [GitHub repository](https://github.com/okta-samples/okta-flask-sample) will be downloaded for you and the environment variables will be configured in the `.okta.env` file. 
+
+**NOTE:** You need to enter the URIs you want to use to redirect. In the CLI these are referred to as Redirect URIs. In your developer dashboard, they're called sign-in redirect URIs. For this walkthrough, we'll use: `http://localhost:3000/authorization-code/callback`. You can use other localhost ports, just be sure to write out the link with the `/authorization-code/callback` added at the end. You also need to add your **Post Logout Redirect URIs**. In the developer dashboard, these are called **Sign-out redirect URIs**. Since we used localhost:3000, the URI will be: `http://localhost:3000/`. 
```

Comment:
```suggestion
**NOTE:** You need to enter the URIs you want to use to redirect. In the CLI these are referred to as Redirect URIs. In your developer dashboard, they're called **Sign-in redirect URIs**. For this walkthrough, we'll use: `http://localhost:3000/authorization-code/callback`. You can use other localhost ports, just be sure to write out the link with the `/authorization-code/callback` added at the end. You also need to add your **Post Logout Redirect URIs**. In the developer dashboard, these are called **Sign-out redirect URIs**. Since we used localhost:3000, the URI will be: `http://localhost:3000/`. 
```